### ISSUE TITLE
.heic / .heif pictures shot by Sony A7SIII and A1 are not shown on Geeqie 1.7.2 under Ubuntu 22.04

### ISSUE THREAD

**Issue by MarkusHB1** [BestImageViewer/geeqie#1100]
### ISSUE TYPE

* Bug Report

### GEEQIE VERSION

```
Geeqie 1.7.2 GTK3
```

```
Download readymade by Ubuntu 22.04
```

**Comment by MarkusHB1** [comment]
Somehow all writing and attachments got lost. Here they come again.
![Bildschirmfoto_2023-06-07_18-54-30](https://github.com/BestImageViewer/geeqie/assets/135862210/e65919c8-9d66-4565-af2a-f81cc6cda0bf)

[A7S05598.HEIC.zip](https://github.com/BestImageViewer/geeqie/files/11680073/A7S05598.HEIC.zip)

I

**Comment by caclark** [comment]
If I open the attached image with `ghex`, I see that the signature is `ftypheix`.

(The problem is that the code looks for `ftypheic`)

Is the code `ftypheix` used for images from both cameras?

**Comment by MarkusHB1** [comment]
Thanks for the superfast reply Colin!
I have no idea, it is done by sony, but I enclosed one by the A1.
[SA107719.HEIC.zip](https://github.com/BestImageViewer/geeqie/files/11680426/SA107719.HEIC.zip)

**Comment by MarkusHB1** [comment]
Thanks Colin & Klaus!
Will there be an update for Ubuntu 22.04?
If Yes, any idea when it will come? I have 7448 pics from my last travel, which I would prefer not to convert :-)

**Comment by caclark** [comment]
The latest version of Geeqie is v2.0.1, but it does not solve your problem as it was released some time ago.
There will be a new version of Geeqie (including this bug fix) in the next few weeks - when a bug or two gets fixed.

There is no need to convert your images if you follow the instructions above. Using this procedure, when you open one of these images  Geeqie loads your image file into `/tmp`, converts it into a .jpg format  file, loads that file into the pixbuf that Geeqie uses internally and then deletes the `/tmp` files.

The extra conversions take a bit of extra time each time you want to look at a file of this type.

[Another method of trying a new version of Geeqie is to use an AppImage. This is a single large file that does not need installation - you just download it and run it. You can get these via geeqie.org. Sometime in the next day or so I will upload a new AppImage that contains this bug fix.]

**Comment by MarkusHB1** [comment]
Thanks for the information Colin!

Do You have any idea when You will publish the update?
Best regards
Markus

Am Fr., 9. Juni 2023 um 16:17 Uhr schrieb Colin Clark <
***@***.***>:

**Comment by caclark** [comment]
I am waiting for a fix for an exiv2 related bug:
https://github.com/Exiv2/exiv2/issues/2638

I think that the problem lies with exiv2 code and not with Geeqie.

At some point I guess I will just make a Geeqie release with the 0.28.0 version of exiv2 excluded from the build.

**Comment by caclark** [comment]
There is a new release. It will take a while to arrive in whichever distribution you are using.

**Comment by SimonDedman** [comment]
Hi Colin sorry to bother you. Is there a trick to getting HEIC to work on Geeqie? I have the latest version (2.2+git20240127-aa162d41, git pulled a second ago) and HEIC images all have the broken image icon. Converting to jpg with [this](https://linuxnightly.com/convert-heif-images-to-jpg-or-png-on-linux/), they work fine. HEIC is ticked as a viewable format in the settings. Any idea what I could be missing? Thanks!

**Comment by caclark** [comment]
My standard xubuntu installation does not show these file. The installed version of libheif is 1.16.2-2

The Geeqie AppImages show these files - but they are compiled on GitHub, which uses libheif 1.12.0

I compiled and installed the latest version of libheif - 1.17.6 - and the files are displayed.

Therefore I assume the problem is in libheif, and it is now fixed.

If you want to mess around with your installation by compiling and installing libheif, that is up to you. I would not recommand that  unless you are confident with this kind of thing.

The AppImage is a short-term possibility until things get straightened out - the README shows how to negate the load time problem with these files.

Another possibility is to use the `Edit/Preferences/Advanced/External Preview` feature - see the Help file.
